resources/js/components/DataGrid.vue

```diff
@@ -0,0 +1,89 @@
+<script setup lang="ts">
+import { onMounted, ref } from 'vue'
+import Pagination from './Pagination.vue'
+
+type singleData = {
+    type: string
+    id: number
+    attributes: {
+        name: string
+        address: string
+        phone: string
+        email: string
+        manager: string
+    }
+    links: {
+        self: string
+    }
+}
+type Data = {
+    data: singleData[]
+    links: {
+        self: string
+        first: string
+        last: string
+        prev: string | null
+        next: string
+    }
+    meta: {
+        current_page: number
+        from: number
+        path: string
+        per_page: number
+        to: number
+    }
+}
+
+interface Props {
```

Comment:
Please implement proper types. Data type was already built (look above).
Use the same type of logic and make type/interface(or mixture of both, if needed) for "fields" as well.